### PR TITLE
API provides functionality to finalize order

### PR DESCRIPTION
**Pull request by DavDan1**
PT: https://www.pivotaltracker.com/story/show/177903772

### User story
```
As a restaurant API
In order for the user to complete their order
I would like to be able to update their existing order to be finalized
```
### Changes suggested
- Create a PUT request to update final order

### CODE REVIEW

File: app/controllers/api/orders_controller.rb [hunk 1]
```diff
@@ -9,17 +9,22 @@ def create
   end
   def update
     order = Order.find(params['id'])
-    menu_item = MenuItem.find(params['menu_item_id']) 
-    new_item = order.order_items.create(menu_item_id: menu_item.id)
+    if params['finalized']
```
Comment (christerforsgren91) [inline app/controllers/api/orders_controller.rb:12]: next time if time allows you, you can refactor the code.

File: app/controllers/api/orders_controller.rb [hunk 2]
```diff
@@ -9,17 +9,22 @@ def create
...
+      order.update(finalized: true)
+      render json: { 
+        message: 'Your order will be able to pick up in 30 minutes!'
```
Comment (offcast-dev) [inline app/controllers/api/orders_controller.rb]: I think you've meant to say:
"Your order will be ready for pick up in 30 minutes!"

File: app/controllers/api/orders_controller.rb (hunk 1 above)
Comment (arunbhalli) [reply app/controllers/api/orders_controller.rb:12]: Thank you for that  suggestion

File: spec/requests/api/api_provides_user_to_finalize_the_order_spec.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      params: {finalized: true},
+      headers: auth_headers
+    end
+
+    it 'is expected to have a 200 status' do
+      expect(response).to have_http_status 200
+    end
+
+    it 'is expected to return a success message' do
+      expect(response_json['message']).to eq 'Your order will be able to pick up in 30 minutes!'
```
Comment (offcast-dev) [inline spec/requests/api/api_provides_user_to_finalize_the_order_spec.rb]: also change it here to correspond

File: app/controllers/api/orders_controller.rb (hunk 2 above)
Comment (DavDan1) [reply app/controllers/api/orders_controller.rb]: I will change it at once.
thank you for your time.

File: spec/requests/api/api_provides_user_to_finalize_the_order_spec.rb (hunk 3 above)
Comment (DavDan1) [reply spec/requests/api/api_provides_user_to_finalize_the_order_spec.rb]: thank you.

**Pull request merged**